daemons/fenced/fenced_history.c

```diff
@@ -347,6 +347,21 @@ stonith_merge_in_history_list(GHashTable *history)
 
         updated = TRUE;
         g_hash_table_iter_steal(&iter);
+
+        if ((op->state != st_failed) &&
+            (op->state != st_done) &&
+            safe_str_eq(op->originator, stonith_our_uname)) {
+            crm_warn("received pending action we are supposed to be the "
+                     "owner but it's not in our records -> fail it");
+            op->state = st_failed;
+            op->completed = time(NULL);
+            /* use -ENODEV to not introduce a new return-code that might
+               trigger unexpected results at other places and to prevent
+               remote_op_done from setting the delegate if not present
+             */
+            stonith_bcast_result_to_peers(op, -ENODEV);
```

Comment:
I think ENODEV will really confuse users trying to investigate a failure. It will be more work to make sure there's no unintended side effects, but I think ECONNRESET would be better.

Reply:
Unfortunately anything else than ENODEV or EHOSTUNREACH will make remote_op_done change the delegate (if not set) to the own node.
Not sure if it is safe to add ECONNRESET there? A grep in pacemaker shows that it doesn't
seem to have an explicit meaning so far. And the exit code it is mapped to (CRM_EX_DISCONNECT) doesn't seem to be used in fenced as well.

Reply:
EHOSTUNREACH would be better than ENODEV.

I think it would be OK to handle ECONNRESET the same way, but another issue is mixed-version clusters.

Reply:
Meaning for now we are actually limited to those 2 because we would get 2 failed entries - first on different nodes and after some syncing probably in the history of a single node.
Unless we are taking a totally different route of course ...

Reply:
Another reason not to use ENODEV is that the controller won't retry a failed stonith action with that result (see tengine_stonith_callback())